lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp

```diff
@@ -72,6 +81,65 @@ class CallOpConversion : public OpConversionPattern<SrcOp> {
   bool explicitOperandTypes;
 };
 
+class Conv2DOpConversion : public OpConversionPattern<tosa::Conv2DOp> {
+
+public:
+  Conv2DOpConversion(MLIRContext *ctx) : OpConversionPattern(ctx) {}
+
+private:
+  LogicalResult
+  matchAndRewrite(tosa::Conv2DOp convOp, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    // tosa conv2D supports adding a bias after the actual convolution. We will
+    // split the convolution with bias into two operations: the convolution (as
+    // an emitc call op) and the addition of the bias (as an tosa.add op).
+    // Therefore we remove bias from the operands of the convolution.
+    operands = operands.drop_back();
+
+    typename tosa::Conv2DOp::Adaptor adaptor(operands);
+    auto ctx = convOp.getContext();
+
+    StringRef funcName = "tosa::conv2d";
+    StringAttr callee = rewriter.getStringAttr(funcName);
+
+    // fail if quantization is requested
+    if (convOp.quantization_info().hasValue()) {
+      return convOp.emitError(
+          "Quantization of tosa.conv2d is currently not supported.");
+    }
+
+    SmallVector<Attribute, 2> args_ = indexSequence(operands.size(), ctx);
+
+    args_.push_back(convOp.pad());
+    args_.push_back(convOp.stride());
+    args_.push_back(convOp.dilation());
+
+    ArrayAttr args = rewriter.getArrayAttr(args_);
```

Comment:
Maybe it's easier to read if you specify the two operands directly via index attributes, i.e.
```c
ArrayAttr args = rewriter.getArrayAttr({
  rewriter.getIndexAttr(0),
  rewriter.getIndexAttr(1),
  convOp.pad(),
  convOp.stride(),
  convOp.dilation()
});
```
(untested)

Reply:
This is awesome!! Not sure if `clang-format` likes it :/

Reply:
Give it a try. Otherwise we can disable clang-format here if it makes the code less readable :)

Reply:
Just saw your comment: sadly `clang-format` doesn't like it.

Reply:
That's why I meant you can turn it off here ;)